tests/e2e/gpu-feature-discovery_test.go

```diff
@@ -135,6 +136,11 @@ var _ = NVDescribe("GPU Feature Discovery", func() {
 
 		// Cleanup before next test run
 		AfterEach(func(ctx context.Context) {
+			// Gather logs
+			if CurrentSpecReport().Failed() {
+				err := common.MustGather(filepath.Join(*LogArtifactDir, f.UniqueName), "gpu-feature-discovery", f.Namespace.Name)
```

Comment:
Nit: Can we add this at a global level?

Reply:
Each test is independent so if done at the global level we would only catch the last run. `AfterEach` is the right place so we run the log collection if a specific test fails.

Each test container creates a namespace and deploys helm on it.